MDANSE/Src/MDANSE/Framework/Jobs/IJob.py

```diff
@@ -307,14 +307,19 @@ def set_up_trajectory(self):
     def run_step(self, index):
         pass
 
-    def preview_output_axis(self):
-        axes = {}
+    def preview_output_axis(self) -> list[tuple[str, Sequence[float], str]]:
+        """Collect the output axis values and unit information from configurators.
+
+        Returns
+        -------
+        dict[str, Sequence[float]]
+            Dictionary of {unit: values} pairs, predicting the data output range.
+        """
+        axes = []
         for configurator in self._configuration.values():
-            preview_method = getattr(configurator, "preview_output_axis", None)
-            if callable(preview_method):
-                axis, unit = preview_method()
-                if axis is not None:
-                    axes[unit] = axis
+            for prediction in configurator.preview_output_axis():
+                if prediction is not None:
+                    axes.append(prediction)
```

Comment:
With the change suggested above, this would become:

```suggestion
        axes = []
        for predictor in self.PREDICTORS:
            configurator = self._configuration[predictor]
            for prediction in configurator.preview_output_axis():
                if prediction is not None:
                    axes.append(prediction)
```